### PR TITLE
Clean up errors

### PR DESCRIPTION
**Pull request by rylev**
This cleans up errors a bit (I'd like to keep going, but this is a good first step):
* This moves the `PolicyError` out of cosmos and to core
* Introduces a catchall error for unimportant errors that are only useful to report to users 
* Moves all parsing errors to its own type

### CODE REVIEW

File: sdk/cosmos/src/responses/query_documents_response.rs [hunk 1]
```diff
@@ -270,28 +270,20 @@ pub struct QueryDocumentsResponseDocuments<T> {
...
             results: q
                 .results
                 .into_iter()
                 .map(|r| match r {
-                    QueryResult::Document(document) => document,
+                    QueryResult::Document(document) => Ok(document),
                     QueryResult::Raw(_) => {
-                        panic!("this should have been caugth by the previous check")
+                        // Bail if there is a raw document
+                        Err(Self::Error::Other("conversion to `QueryDocumentsResponseDocuments` failed because at lease one element is raw".into()))
```
Comment (heaths) [inline sdk/cosmos/src/responses/query_documents_response.rs:283]: Re: "raw"

In the context of Cosmos, will this reference make sense to a new customer? Does it require clarification?

**Review by heaths — APPROVED** [review]

File: sdk/core/src/pipeline.rs [hunk 2]
```diff
@@ -81,9 +81,10 @@ impl Pipeline {
         self.http_client.as_ref()
     }
 
-    pub async fn send(&self, ctx: &mut Context, request: &mut Request) -> PolicyResult<Response> {
+    pub async fn send(&self, ctx: &mut Context, request: &mut Request) -> Result<Response, Error> {
```
Comment (MindFlavor) [inline sdk/core/src/pipeline.rs:84]: Much more idiomatic! I love it! 👍

**Review by MindFlavor — APPROVED** [review]
Nice job!

File: sdk/cosmos/src/responses/query_documents_response.rs (hunk 1 above)
Comment (rylev) [reply sdk/cosmos/src/responses/query_documents_response.rs:283]: Probably not, but I do think this is a strictly better version than what we had. Any suggestions on how to make it even more clear are welcome 😊

**Pull request merged**